src/debugger/Debugger.cpp

```diff
@@ -206,10 +206,19 @@ bool Debugger::doEval(ExecutionState* state, ByteCodeBlock* byteCodeBlock, uint8
 
     if (type == ESCARGOT_MESSAGE_EVAL_8BIT) {
         str = new Latin1String(data, size);
-    } else {
+    } else if (type == ESCARGOT_MESSAGE_EVAL_16BIT) {
         str = new UTF16String((char16_t*)data, size / 2);
+    } else if (type == ESCARGOT_DEBUGGER_CLIENT_SOURCE_8BIT) {
+        uint32_t sourceNameLen = strlen(data);
```

Comment:
A security question: there is no guarantee, that a data received from network has \0. I think `memchr` is safer.